### DISCUSSION TITLE
Expose `initialData` in `useContractRead` options

### DISCUSSION THREAD

**Discussion by greg-schrammel** [wevm/wagmi#1914] [Ideas]
yo, thoughts on exposing the `initialData` react query option?

my current setup is this
a util contract fetches all available markets with it's latest price,   
a market is selected and then I wanna watch the price on the route selected market,  

would be cool to have the initial data from the other contract call already there
another way I see is set with the `queryClient.setQueryData`, but initialData feels better

```ts
  const market = useRouteMarket()

  const { data: price } = useContractRead({
    abi: marketABI,
    address: market?.address,
    functionName: 'assetPrice',
    select: ({ price }) => FixedNumber.fromValue(price),
    watch: true,
    cacheOnBlock: true,
    initialData: { price: market?.price }, // <-- wanna do that :(
  })
```

**Comment by tmm** [comment]
For `wagmi@1`, we are going to streamline our React Query usage so these things are mostly automatic.

**Reply by snorkypie** [reply]
Was this solved with `wagmi@1`?

**Reply by tmm** [reply]
We are waiting until React Query v5 is released to make some updates.
